exasol_sphinx_github_pages_generator/generate_index.py

```diff
@@ -0,0 +1,157 @@
+            current_branch = run(["git", "rev-parse", "--abbrev-ref", "HEAD"], capture_output=True, text=True,
+                                 check=True)
+            print(f"current_branch {current_branch.stdout} in find_index_worktree_path {find_index_worktree_path}.")
+            # does not work if release branch name contains slash
+            release_list = (name for name in os.listdir(find_index_worktree_path)
+                            if os.path.isdir(os.path.join(find_index_worktree_path, name)))
+
+            release_list_dicts = generate_release_dicts(release_list, source_branch, target_worktree)
+            run(["git", "worktree", "remove", "--force", find_index_worktree_path], check=True)
+
+    else:
+        release_list_dicts = generate_release_dicts([], source_branch, target_worktree)
+
+    return release_list_dicts
+
+
+def generate_release_index(target_branch: str, target_worktree: Path, source_branch: str, target_branch_exists_remote: bool) -> None:
+    """
+    Generates a release index file from a given target_branch into the target_worktree.
+    Uses the "furo" theme for the generated release index file.
+    Source_branch will be called "latest" in the resulting release index.
+    Also adds the template file to "_sources".
+    Aborts if not currently in given source_branch.
+
+    :param target_branch: The branch the generated documentation should end up on.
+    :param target_worktree: Worktree/path all generated project documentation is put into.
+    :param source_branch: The branch we are currently generating the documentation for.
+    :param target_branch_exists_remote: Bool representing whether the target_branch already exist in the remote
+        repository.
+
+    """
+
+    print("s_branch " + source_branch)
+    local_source_branch_commit_id = run(["git", "rev-parse", source_branch], capture_output=True,
+                                        text=True)
+    if local_source_branch_commit_id.returncode != 0:
+        sys.exit(f"{source_branch} not currently checked out. Please Check out branch {source_branch}"
+                 f" before calling generate_release_index.")
+    # remove slashes from branch-name, this makes parsing the release-names for the release-index much easier
+    simple_source_branch_name = source_branch.replace("/", "-")
+    env = Environment(
+        loader=PackageLoader("exasol_sphinx_github_pages_generator"),
+        autoescape=select_autoescape()
+    )
+    template = env.get_template("index_template.html.jinja2")
+    releases = get_releases(target_branch, target_branch_exists_remote, simple_source_branch_name, target_worktree)
+    with open(f"{target_worktree}/index.html", "w+") as file:
+        file.write(template.render(meta_list=[], releases=releases, footer=[]))
+    run(["ls", "-la", ".."], check=True)
+
+    generator_init_path = inspect.getfile(exasol_sphinx_github_pages_generator)
+    sources_dir = f"{os.path.dirname(generator_init_path)}/templates"
```

Comment:
if you need files from the packages use https://pypi.org/project/importlib-resources/. Your approach doesn't work everywhere

Reply:
changed

Reply:
You use `generator_init_path` later still with filesystem function, everything, you need to use for all paths you derive from it, the importlib-ressources to get the files. You also need to read the content of the files with  importlib-ressources and write them to disk.